### ISSUE TITLE
No support for optional chaining

### ISSUE THREAD

**Issue by pmrotule** [hedefalk/atom-vue#97]
Currently no support for optional chaining: https://babeljs.io/docs/en/babel-plugin-proposal-optional-chaining

Since it is now possible to use it with Babel, it would be great to be able to use it in Vue component.

<img width="650" alt="image" src="https://user-images.githubusercontent.com/10983258/51443299-05b0b280-1ce7-11e9-9921-d2ca9e62832c.png">

**Comment by aidistan** [comment]
Hi @pmrotule. Actually this issue has no relation to `language-vue` , since the support for optional chaining your need should be provided by the javascript language package you used. Give any package providing `source.js.jsx` grammar (like [this](https://atom.io/packages/language-babel)) a try!

**Comment by pmrotule** [comment]
@aidistan I'm using the core package `language-javascript` which fully support optional chaining and indeed, I don't have any highlighting issue in normal javascript files. Moreover, the javascript itself within the script tag of the Vue single-file component has no highlighting issue or syntax error, it only breaks from the closing tag `</script>`. This must be an issue coming from `language-vue`.

<img width="756" alt="image" src="https://user-images.githubusercontent.com/10983258/51517628-a99b7a80-1e1b-11e9-8465-c8dec87ee902.png">

**Comment by aidistan** [comment]
@pmrotule You're right. It's comfirmed. I'm quite occupied in these days. Will look into it ASAP.

**Comment by pmrotule** [comment]
@aidistan After looking into it a bit deeper, I realized that the problem is not coming from this package, but from embedded script. I mentioned that it was fully supported by `language-javascript`, but it actually only works properly in javascript files and not within a `<script>` tag in HTML.

`language-vue` use `source.js.embedded.html` and therefore has the same issue as any other html files. I reported it to `language-javascript`. Sorry about the confusion.
